Update docs and error-handling for Celery worker integration. (#635)
nautobot/nautobot: nautobot/docs/installation/wsgi.md

> # Deploying Nautobot: Web Service and Worker

Reviewer: Would it make sense to rename this file to something other than `wsgi.md` - perhaps `services.md` or `systemd.md` or such?

Author: Yeah good call. Thanks.

Additionally, certain Nautobot features (including Git repository synchronization, Webhooks, Jobs, etc.) depend on the presence of Nautobot's [Celery](https://docs.celeryproject.org/en/stable/) background worker process, which is not automatically started with Nautobot and is run as a separate service.

This document will guide you through setting up uWSGI and establishing Nautobot web and Celery worker services to run on system startup.

### Web Service

Nautobot includes a `nautobot-server start` management command that directly invokes uWSGI. This command behaves exactly as uWSGI does, but allows us to maintain a single entrypoint into the Nautobot application.

```no-highlight
$ nautobot-server start --help
```

### Worker Service

Nautobot requires at least one worker to consume background tasks required for advanced background features. A `nautobot-server celery` command is included that directly invokes Celery. This command behaves exactly as the Celery command-line utility does, but launches it through Nautobot's environment to share Redis and database connection settings transparently.

```no-highlight
$ nautobot-server celery --help  
```

!!! important

> 	Prior to version 1.1.0, Nautobot utilized RQ as the primary background task worker. As of Nautobot 1.1.0, RQ is now *deprecated*. RQ and the `@jobs` decorator for custom Jobs is still supported for now, but will no longer be documented, and support for RQ will be removed in a future release.

Reviewer: ```suggestion
	Prior to version 1.1.0, Nautobot utilized RQ as the primary background task worker. As of Nautobot 1.1.0, RQ is now *deprecated*. RQ and the `@job` decorator for custom tasks are still supported for now, but will no longer be documented, and support for RQ will be removed in a future release.
```

## Setup systemd

We'll use `systemd` to control both uWSGI and Nautobot's background worker processes.

!!! warning
    The following steps must be performed with root permissions.

### Nautobot service

> First, we'll setablish the `systemd` unit file for the Nautobot web service. Copy and paste the following into `/etc/systemd/system/nautobot.service`:

Reviewer: ```suggestion
First, we'll establish the `systemd` unit file for the Nautobot web service. Copy and paste the following into `/etc/systemd/system/nautobot.service`:
```

WantedBy=multi-user.target
```

> ### Nautobot  Worker Service

Reviewer: ```suggestion
### Nautobot Worker Service
```

$ sudo systemctl daemon-reload
```

> Then, start the `nautobot` and `nautobot-worker`services and enable them to initiate at boot time:

Reviewer: ```suggestion
Then, start the `nautobot` and `nautobot-worker` services and enable them to initiate at boot time:
```
